docs/reference/workflows/ecs/concepts/temporary-components.md

```diff
@@ -0,0 +1,18 @@
+<%(TOC)%>
+
+# ECS: Temporary components
+
+A temporary component is a type of component which only exists for a single frame. This pattern is useful when you want to represent transient events as an ECS component, for example, the [`OnConnected`]({{urlRoot}}/api/core/on-connected) component which denotes that your worker has connected to the SpatialOS Runtime as described in the [ECS: Worker entity documentation]({{urlRoot}}/reference/workflows/ecs/worker-entity).
+
+We provide you with the [`Improbable.Gdk.Core.RemoveAtEndOfTick`]({{urlRoot}}/api/core/remove-at-end-of-tick-attribute) attribute, which can be applied to any ECS component to make it a temporary component.
+
+If a temporary component is added to an entity, the component will be removed at the end of the update loop. This means that when you use temporary components, you must consider the ordering of your systems carefully. Any system which runs logic predicated on the temporary component should be run after the temporary component is added and before it is removed.
```

Comment:
```suggestion
If a temporary component is added to an entity, the GDK removes the component at the end of the update loop. This means that when you use temporary components, you must consider the ordering of your systems carefully. Any system which runs logic predicated on the temporary component should be run after the temporary component is added and before it is removed.
```

should we also mention in which system they get removed?

and also an example of how to order such a system

Reply:
we used to have that but it became an incredibly verbose example of just how to add a component

Reply:
Yeah we aren't teaching users how to use the ECS (just as we don't teach users how to use GameObjects & MonoBehaviours)